libp2p/wire.nim

```diff
@@ -79,28 +87,46 @@ proc createStreamServer*[T](ma: MultiAddress,
                             child: StreamServer = nil,
                             init: TransportInitCallback = nil): StreamServer =
   ## Create new TCP stream server which bounds to ``ma`` address.
-  var address = initTAddress(ma)
-  if address.family in {AddressFamily.IPv4, AddressFamily.IPv6}:
-    if ma[1].tryGet().protoCode().tryGet() != multiCodec("tcp"):
-      raise newException(TransportAddressError, "Incorrect address type!")
-  result = createStreamServer(address, cbproc, flags, udata, sock, backlog,
+  if not(RTRANSPMA.match(ma)):
+    raise newException(TransportAddressError, "Incorrect address type!")
```

Comment:
This is incorrect exception type and message, because address could be absolutely correct (for example "/ip4/127.0.0.1/udp/13000`) but it would be unsupported here, so i think message should be different.
Also it is better not use `TransportAddressError` because this is `nim-chronos` defined Exception type and used to signal about incorrect addresses. So maybe its better to rename the exception type too.

Reply:
Interesting, this is a copy paste from you original change in https://github.com/status-im/nim-libp2p/pull/247/files#diff-7831cb8a03a4c3de33123e27a62ca63d25b578b618ea8623e72f2135ed89ea66R74 (just making sure, maybe it made sense back then but not anymore). So you suggest I remap the exception to something else or not use it at all?

Reply:
I think libp2p as a high level network framework should have its own exception types, because for `chronos` this exception `TransportAddressError` means that address is incorrect, but for `libp2p` its meaning is different, its not `incorrect` but `not-supported`.